api/matlab/arrus/DuplexDisplay.m

```diff
@@ -12,26 +12,82 @@
         showTimes
         dynamicRange
         powerThreshold
+        
+        cineLoop
+        cineLoopLength
+        cineLoopIndex
+        
+        persistence
+        
+        bmodeTgc
+        bmodeAutoTgcResp
     end
     
     methods 
-        function obj = DuplexDisplay(proc, dynamicRange, powerThreshold, subplotEnable)
+        function obj = DuplexDisplay(varargin)
+            
+            % Input parser
+            dispParParser = inputParser;
+            addParameter(dispParParser, 'reconstructionObject', []);
+            addParameter(dispParParser, 'dynamicRange', [0 80]);
+            addParameter(dispParParser, 'powerThreshold', -inf);
+            addParameter(dispParParser, 'subplotEnable', false);
+            addParameter(dispParParser, 'cineLoopLength', 1);
+            addParameter(dispParParser, 'persistence', 1);
+            addParameter(dispParParser, 'bmodeTgc', 0);
+            addParameter(dispParParser, 'bmodeAutoTgcResp', 0);
+            parse(dispParParser, varargin{:});
             
-            if nargin < 4
-                subplotEnable = false;
+            proc             = dispParParser.Results.reconstructionObject;
+            dynamicRange     = dispParParser.Results.dynamicRange;
+            powerThreshold   = dispParParser.Results.powerThreshold;
+            subplotEnable    = dispParParser.Results.subplotEnable;
+            cineLoopLength   = dispParParser.Results.cineLoopLength;
+            persistence      = dispParParser.Results.persistence;
+            bmodeTgc         = dispParParser.Results.bmodeTgc;
+            bmodeAutoTgcResp = dispParParser.Results.bmodeAutoTgcResp;
+            
+            if isempty(proc)
+                error("ARRUS:IllegalArgument", "reconstructionObject is an obligatory input.");
+            elseif ~isa(proc,'Reconstruction')
+                error("ARRUS:IllegalArgument", "reconstructionObject must be of class Reconstruction.");
             end
             
-            if nargin < 3
-                powerThreshold = -inf;
+            if ~all(size(dynamicRange) == [1 2]) || ~isnumeric(dynamicRange)
+                error("ARRUS:IllegalArgument", "dynamicRange must be a 2-element numerical vector: [min max]");
             end
             
-            if nargin < 2
-                dynamicRange = [0 60];
-            else 
-                if ~all(size(dynamicRange) == [1 2])
-                    error("ARRUS:IllegalArgument", ...
-                        "Invalid dimensions of dynamic range vector, should be: [min max]")
-                end
+            if ~isscalar(powerThreshold) || ~isnumeric(powerThreshold)
```

Comment:
Consider moving validation functions into addParameter() method.

Reply:
In future development - yes, for now it can stay this way.

Reply:
Done